### PR TITLE
implemented py-distance-class

### PR DESCRIPTION
**Pull request by mariiahorbova**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,65 @@
...
+            return self.km > other
+        return self.km > other.km
+
+    def __lt__(self, other: Any) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km < other
+        return self.km < other.km
+
+    def __ge__(self, other: Any) -> bool:
+        if isinstance(other, (int, float)):
```
Comment (LiudmylaKulinchenko) [inline app/main.py:58]: You should use previously created magic method here to simplify your code (call methods using operands)

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,65 @@
...
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Any:
+        if isinstance(other, (int, float)):
+            distance = Distance(self.km + other)
```
Comment (LiudmylaKulinchenko) [inline app/main.py:16]: The additional variable is redundant here

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,65 @@
...
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> Any:
```
[the remark this answers is not in the document]
Comment (mariiahorbova) [reply app/main.py:14]: done

File: app/main.py [hunk 4]
```diff
@@ -1,3 +1,65 @@
...
+        if isinstance(other, (int, float)):
```
[the remark this answers is not in the document]
Comment (mariiahorbova) [reply app/main.py]: done

File: app/main.py (hunk 1 above)
Comment (mariiahorbova) [reply app/main.py:57]: done

File: app/main.py (hunk 2 above)
Comment (mariiahorbova) [reply app/main.py]: don't quite understand the comment

Comment (mariiahorbova) [reply app/main.py]: done

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,61 @@
...
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        else:
```
Comment (YuriiKindrat) [inline app/main.py:17]: why do u need else here?

File: app/main.py [hunk 6]
```diff
@@ -1,3 +1,61 @@
...
+            self.km = self.km + other
+            return self
+
+    def __iadd__(self, other: Distance | int) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km = self.km + other
+            return self
+        else:
+            self.km += other.km
+            return self
```
Comment (YuriiKindrat) [inline app/main.py:27]: u are returning the same thing every time, could u use only one return here?

File: app/main.py [hunk 7]
```diff
@@ -1,3 +1,61 @@
...
+    def __truediv__(self, number: float | int) -> Distance:
+        return Distance(round(self.km / number, 2))
+
+    def __eq__(self, other: Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km == other
+        return self.km == other.km
+
+    def __ne__(self, other: Distance) -> bool:
+        return not self.km.__eq__(other.km)
```
Comment (YuriiKindrat) [inline app/main.py:41]: no need to call method in that way, u can use operator for this `=`

File: app/main.py [hunk 8]
```diff
@@ -1,3 +1,61 @@
...
+
+    def __gt__(self, other: Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return not self.km.__eq__(other) and not self.km.__lt__(other)
+        return not self.km.__eq__(other.km) and not self.km.__lt__(other.km)
```
Comment (YuriiKindrat) [inline app/main.py:46]: same here

File: app/main.py [hunk 9]
```diff
@@ -1,3 +1,61 @@
...
+
+    def __ge__(self, other: Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return not self.km.__lt__(other)
+        return not self.km.__lt__(other.km)
+
+    def __le__(self, other: Distance) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km.__eq__(other) or self.km.__lt__(other)
+        return self.km.__eq__(other.km) or self.km.__lt__(other.km)
```
Comment (YuriiKindrat) [inline app/main.py:61]: same

Comment (YuriiKindrat) [reply app/main.py:61]: u can implement a le/ge mthods in simpler way
hint: `>= is not <`

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]
also, check ur type annotations for other in eq, gt, lt, etc. methods, is other will always be a Distance instance ?

File: app/main.py [hunk 10]
```diff
@@ -1,3 +1,59 @@
...
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | int) -> Distance:
```
Comment (LiudmylaKulinchenko) [inline app/main.py:14]: `other` could also be `float` instance. Fix annotations, please 🌺

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 11]
```diff
@@ -1,3 +1,59 @@
...
+
+    def __lt__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km < other
+        return self.km < other.km
+
+    def __ge__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, (int, float)):
+            return not self.km < other
+        return not self.km < other.km
```
Comment (YuriiKindrat) [inline app/main.py:54]: no need to make check for instance here, u are already check this in lt method, that u are using here

File: app/main.py [hunk 12]
```diff
@@ -1,3 +1,59 @@
...
+
+    def __le__(self, other: Distance | int | float) -> bool:
+        if isinstance(other, (int, float)):
+            return self.km == other or self.km < other
+        return self.km == other.km or self.km < other.km
```
Comment (YuriiKindrat) [inline app/main.py:59]: same is here, also u can make a backward condition as for ge method

**Review by YuriiKindrat — CHANGES_REQUESTED** [review]

**Review by YuriiKindrat — APPROVED** [review]

**Pull request closed without merging**